source/extensions/filters/http/ext_proc/processor_state.h

```diff
@@ -139,16 +142,17 @@ class ProcessorState : public Logger::Loggable<Logger::Id::ext_proc> {
   virtual void injectDataToFilterChain(Buffer::Instance& data, bool end_stream) PURE;
   virtual uint32_t bufferLimit() const PURE;
 
+  ChunkQueue& chunkQueue() { return chunk_queue_; }
   // Move the contents of "data" into a QueuedChunk object on the streaming queue.
   void enqueueStreamingChunk(Buffer::Instance& data, bool end_stream, bool delivered);
   // If the queue has chunks, return the head of the queue.
-  absl::optional<QueuedChunkPtr> dequeueStreamingChunk(bool undelivered_only) {
-    return chunk_queue_.pop(undelivered_only);
+  absl::optional<QueuedChunkPtr> dequeueStreamingChunk(bool undelivered_only,
```

Comment:
interface nit: not your code.
I think `absl::optional` is redundant: nullptr can represent empty state.

Reply:
Thanks for the comments!  Understand the point. But the purpose of this PR is to fix the chunk management issue.  Remove absl::optional will trigger corresponding changes in a few places. We can have a separate PR to do such cleanups if necessary.